sovryn-token-bridge/federator/src/lib/Federator.js

```diff
@@ -415,4 +456,35 @@ module.exports = class Federator {
     async _getCurrentBlockNumber() {
         return this.mainWeb3.eth.getBlockNumber();
     }
+
+    async signTransaction({ blockNumber, id }) {
+        const logs = await this.mainBridgeContract.getPastEvents('Cross', {
+            fromBlock: blockNumber,
+            toBlock: blockNumber,
+            filter: { id },
+        });
+
+        if (logs.length !== 1) throw new Error('Invalid return when searching for event');
+
+        const { _tokenAddress, _amount, _to, _symbol, _decimals, _granularity, _userData } =
+            logs[0].returnValues;
+        const { blockHash, transactionHash, logIndex } = logs[0];
+        const txId = await this.federationContract.methods
+            .getTransactionIdU(
+                _tokenAddress,
+                _to,
+                _amount,
+                _symbol,
+                blockHash,
+                transactionHash,
+                logIndex,
+                _decimals,
+                _granularity,
+                _userData || []
+            )
+            .call();
+
+        const wallet = new ethers.Wallet(this.config.privateKey);
+        return await wallet.signMessage(ethers.utils.arrayify(txId));
```

Comment:
I'm not 100% sure about this, but we might want to have other data besides the `transactionIdU` to sign, to prevent replay attacks. This also affects the smart contract. We could go over this with Haku, to be sure.